Cargo.toml

```diff
@@ -1,6 +1,7 @@
 [workspace]
 members = ["relay", "relay-*", "tools/*"]
 default-members = ["relay"]
+resolver = "2"
```

Comment:
We pretty much use the 2021 Rust edition, which according to [docs](https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html) uses `resolver=2` by default:

> Starting in Rust 2021, this will be the default. That is, writing edition = "2021" in Cargo.toml will imply resolver = "2".

Am I missing something? Or what are the benefits of typing it explicitly?

Reply:
Because of the warning 
```
warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
```

Reply:
Would it be easier to upgrade `relay-kafka` to `edition = 2021`? Or would that cause other problems?

Reply:
`relay-kafka` is upgraded, see the changes in `relay-kafka/Cargo.toml`

Reply:
> virtual workspaces default to `resolver = "1"

Sorry, should have read this. I thought the fallback to resolver = 1 happened because of the kafka crate.

Reply:
Also missed it hehe. Is this happening due to external crates using older editions?

Reply:
See the [`docs on resolvers`](https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions)

Reply:
@iker-barriocanal  also the docs you posted are describing exactly the same https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html#details

Reply:
> If you are using a virtual workspace, you will still need to explicitly set the resolver field in the [workspace] definition if you want to opt-in to the new resolver.

Should have read the whole section 😄